rustls/src/compress.rs

```diff
@@ -226,6 +234,184 @@ mod feat_brotli {
 #[cfg(feature = "brotli")]
 pub use feat_brotli::{BROTLI_COMPRESSOR, BROTLI_DECOMPRESSOR};
 
+/// An LRU cache for compressions.
+///
+/// The prospect of being able to reuse a given compression for many connections
+/// means we can afford to spend more time on that compression (by passing
+/// `CompressionLevel::Amortized` to the compressor).
+#[derive(Debug)]
+pub enum CompressionCache {
```

Comment:
Good idea vs using a trait like I originally suggested :+1: In retrospect that feels like the wrong suggestion since the only implementations are meant to be internal and there's a fixed number.